rescan.go

```diff
@@ -573,25 +538,51 @@ func rescan(chain ChainSource, options ...RescanOption) error {
 		// If the filter is nil, then this either means that we don't
 		// have any peers to fetch this filter from, or the peer(s) that
 		// we're trying to fetch from are in the progress of a re-org.
+		// In any case, we'll attempt to retry fetching the filter until
+		// success or we determine it belongs to a stale block.
```

Comment:
To simplify, we would just loop here until we have the filter.

Reply:
I don't think we can loop here cause we need a signal to stop fetching the filter if the corresponding block gets reorged.

Reply:
wouldn't we get a block disconnected notification in that case?

Reply:
Yeah but if we loop here we won't process any further notifications. Even if we do, it's possible we receive a connected notification instead, which we'd need to stash anyway.

Reply:
I'm okay with keeping the current design, and we can circle back to this later if needed.